### ISSUE TITLE
terminal content flickers when writing reverse background text

### ISSUE THREAD

**Issue by rshouker** [microsoft/vscode#190093]
Type: <b>Bug</b>

This is on Windows 10. Using theme: Cobalt2 https://marketplace.visualstudio.com/items?itemName=wesbos.theme-cobalt2

install aider on wsl: https://github.com/paul-gauthier/aider#getting-started
requires openai api key.

in a terminal enter wsl and run aider with a source code file.
ask it to change something in a source code.
it will show changes on a white background, and it would flicker while aider is slowly writing the code.

VS Code version: Code 1.81.0 (6445d93c81ebe42c4cbd7a60712e0b17d9463e97, 2023-08-02T12:37:13.485Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1992)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.84GB (6.19GB free)|
|Process Argv|--crash-reporter-id 8ec5de7e-3755-421e-b214-9da66c35478d|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (39)</summary>

Extension|Author (truncated)|Version
---|---|---
code-gnu-global|aus|0.2.2
vscode-pandoc|Chr|0.4.3
doxdocgen|csc|1.4.0
overtype|DrM|0.5.0
copilot|Git|1.101.317
copilot-labs|Git|0.14.884
vscode-pull-request-github|Git|0.70.0
vscode-drawio|hed|1.6.6
better-cpp-syntax|jef|1.17.2
vsc-python-indent|Kev|1.18.0
openscad-language-support|Lea|1.2.5
vscode-language-babel|mgm|0.0.39
nc-gcode|ML2|0.14.1
vscode-docker|ms-|1.26.0
autopep8|ms-|2023.6.0
python|ms-|2023.14.0
vscode-pylance|ms-|2023.8.10
remote-containers|ms-|0.304.0
remote-ssh|ms-|0.102.0
remote-wsl|ms-|0.81.0
vscode-remote-extensionpack|ms-|0.24.0
cmake-tools|ms-|1.14.34
cpptools|ms-|1.17.0
cpptools-extension-pack|ms-|1.3.0
remote-explorer|ms-|0.4.1
remote-server|ms-|1.4.0
platformio-ide|pla|3.3.1
java|red|1.21.0
vscode-qt-for-python|sea|7.4.0
cmake|twx|0.0.17
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
vscode-java-debug|vsc|0.53.0
vscode-java-dependency|vsc|0.23.1
vscode-java-pack|vsc|0.25.13
vscode-java-test|vsc|0.39.1
vscode-maven|vsc|0.42.0
debug|web|0.26.1
markdown-all-in-one|yzh|3.5.1

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30792226
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30803845
3biah626:30602489
89544117:30613380
2i9eh265:30646982
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
ecj1e332:30736112
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
gsofa:30804715
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
e537b577:30795824
dsvsc013:30795093
dsvsc014:30804076

```

</details>

**Comment by Tyriar** [comment]
Please provide a screenshot or [a gif](https://gifcap.dev/) illustrating the problem (attaching images won't work via email, you need to go to github.com). See [this wiki page](https://github.com/microsoft/vscode/wiki/Terminal-Issues) for more general info on how to report actionable terminal issues.

/label "info-needed" terminal

**Comment by rshouker** [comment]
here is a [video ](https://www.youtube.com/watch?v=Zy4FRrIlhCs)
I failed to screen capture, so i use my smartphone camera, sorry about the noisy audio.

**Comment by Tyriar** [comment]
Huh, that's strange. Can you reproduce this when you run the same thing in Windows Terminal?

**Comment by rshouker** [comment]
@Tyriar, no, it doesn't have the flickering effect in Windows Terminal; it works just fine there.
